### PR TITLE
Make localized strings accessible after deserialization

### LINKED ISSUE #829: Localized strings are not initialized for modules and hardpoints read from shipyard.json
## EDDI version in which issue found
3.0.3, with VoiceAttack

## Steps to reproduce
Reported by Nepomuk on [Frontier Forums](https://forums.frontier.co.uk/showthread.php/387955-EDDI-2-4-Bring-your-cockpit-to-life?p=6981722&viewfull=1#post6981722):
```
Quick question, why is the variable {TXT: ship compartment n module}: the name of the module in this slot ,
no longer "filled"? https://github.com/EDCD/EDDI/wiki/Vo...ship-variables
```
Either in Cottle or in VoiceAttack, inspect the value of your ship module's localizedNames after they are read from shipmonitor.json. 

## Expected
localizedNames should be available

## Observed
localizedNames are null

## Investigation
Derived string values like localizedName are not written when shipmonitor.json is deserialized (ref.  `ShipMonitor.cs`, `readShips()` method).

### PR DESCRIPTION
**Pull request by Tkael**
Fix #829.

### CODE REVIEW

**Comment by richardbuckle** [comment]
I only have a web browser here so can't run the code. Could you give me an overview of how the fix works please? On the face of it, it seems odd that derived properties with no setters should be JsonRequired -- it the Json being read by some third party?

**Comment by Tkael** [comment]
No, but standard deserialization doesn't seem to populate the localizedName, basename, or invariantName properties without the tags. I'm not really super happy with this solution since I still think we shouldn't need to serialize more than the edname. 

I've also tried using OnDeserialized but I'm not sure how to apply it in this case.

**Comment by richardbuckle** [comment]
Hmm. I may need to read up on how derived properties work, and maybe do some experiments. My understanding was that the `get` method is called each time, so that they change with the properties they are derived from. If this is not the case then we'd better know about it!

**Comment by richardbuckle** [comment]
localizedName is particularly problematic as what happens if you serialise in FR then deserialize in DE? This is the reason MS tell us only to persist keys and invariant values. Hmm.

**Comment by Tkael** [comment]
Note that this problem does not occur after a Frontier API update. Only after reading from shipmonitor.json

File: Tests/ShipTests.cs [hunk 1]
```diff
@@ -105,10 +110,6 @@ public void TestShipScenario1()
             Ship sidewinder;
             Ship courier;
 
-            // Set ourselves as in beta to stop sending data to remote systems
```
Comment (Tkael) [inline Tests/ShipTests.cs]: Since we do this when we initiate the tests, we don't need to do it again.

**Comment by Tkael** [comment]
May also fix #845.

**Pull request merged**